src/averageAges.js

```diff
@@ -55,7 +67,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const ageDifference = [];
+
+  const children = people.filter(child => people.some(person =>
+    child.mother === person.name
+    && (!onlyWithSon || child.sex === 'm')
+  )
+  );
+
+  children.map(function(child) {
```

Comment:
map is returning you a new array, so no need of pushing to a new array, just assign map to a new variable

Reply:
and it's better to use arrow function as callback